dds/DCPS/DataWriterImpl.h

```diff
@@ -572,6 +607,84 @@ class OpenDDS_Dcps_Export DataWriterImpl
   virtual SendControlStatus send_control(const DataSampleHeader& header,
                                          Message_Block_Ptr msg);
 
+  bool skip_serialize_;
+
+  /**
+   * Used to hold the encoding and get the buffer sizes needed to store the
+   * results of the encoding.
+   */
+  class EncodingMode {
+  public:
+    EncodingMode()
+    : valid_(false)
+    , header_size_(0)
+    {
+    }
+
+    EncodingMode(const TypeSupportImpl* ts, Encoding::Kind kind, bool swap_the_bytes)
+    : valid_(true)
+    , encoding_(kind, swap_the_bytes)
+    , header_size_(encoding_.is_encapsulated() ? EncapsulationHeader::serialized_size : 0)
+    , bound_(ts->serialized_size_bound(encoding_))
+    , key_only_bound_(ts->key_only_serialized_size_bound(encoding_))
+    {
+    }
+
+    bool valid() const
+    {
+      return valid_;
+    }
```

Comment:
```suggestion
    bool valid() const { return valid_; }
```
These are so short they can be 1-liners... but it's down to personal preference so either way is fine

Reply:
I mostly do multiple lines to allow for expansion latter, but not always.

Reply:
> I mostly do multiple lines to allow for expansion later, but not always.

Changing it later is fine too, not much added complexity in reviewing that (future) diff.